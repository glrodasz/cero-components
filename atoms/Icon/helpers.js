const ICONS_ASSETS_PATH = '/icons'

const iconMap = {
  arrowRight: 'arrow-right',
  angleDown: 'angle-down',
}

const iconSize = {
  sm: 15,
  md: 26,
  lg: 30,
  xl: 40,
}

export const mapType = (type) => `${ICONS_ASSETS_PATH}/${iconMap[type]}.svg`
export const mapSize = (size) => iconSize[size]
