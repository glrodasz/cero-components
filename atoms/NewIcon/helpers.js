import React from 'react'

class Icon {
  constructor(viewBox, svg) {
    this.viewBox = viewBox
    this.svg = <>{svg}</>
  }
}

export const iconsDictionary = {
  'arrow-right': new Icon(
    '0 0 19 20',
    (
      <>
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M8.65792 19.316L7.34192 18L15.6849 9.658L7.34192 1.316L8.65792 0L18.3149 9.658L8.65792 19.316Z"
          fill="#987CE6"
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M0 8.65796H17V10.658H0V8.65796Z"
          fill="#987CE6"
        />
      </>
    )
  ),
  'angle-down': new Icon(
    '0 0 15 9',
    (
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M7.82633 8.70623L0.882568 1.54063L1.89405 0.496826L7.82633 6.61863L13.7586 0.496826L14.7701 1.54063L7.82633 8.70623Z"
        fill="#46596F"
      />
    )
  ),
  default: null,
}

const iconSize = {
  sm: 12,
  md: 16,
  lg: 24,
  xl: 32,
}

export const mapSize = (size) => iconSize[size]
