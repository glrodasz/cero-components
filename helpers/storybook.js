/* eslint-disable react/display-name, react/jsx-key */
import React from 'react'
import { getClasses } from './styles'

export const getTemplate = (Component, styles) => (args) => {
  const allProps = { ...Component.defaultProps, ...args }
  return <Component {...args} getStyles={getClasses(styles)(allProps)} />
}

export const getListTemplate = (Component, styles) => ({ items, ...args }) =>
  items.map((item, index) => {
    const allProps = { ...Component.defaultProps, ...args, ...item }
    return (
      <Component
        key={index}
        {...args}
        {...item}
        getStyles={getClasses(styles)(allProps)}
      />
    )
  })

export const getOptionsArgTypes = (options) => ({
  description: '**options:**',
  table: {
    type: { summary: options.map((option) => `'${option}'`).join('|') },
  },
  control: { type: 'select', options },
})
