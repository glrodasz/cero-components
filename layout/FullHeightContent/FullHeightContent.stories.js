import { FullHeightContent, styles } from '.'

import { getTemplate } from '../../helpers/storybook'

const Template = getTemplate(FullHeightContent, styles)

export default {
  title: 'Layout/FullHeightContent',
  component: FullHeightContent,
  args: {
    isVisible: true,
    content:
      'Mice defiling required Bilbo sailing unspoiled this hunting. 14th refuge knowing breath Middle-earth sack demon children?',
    footer:
      'Ambition living aid indeed mutton suffered l failing kindness Ã‰othain? Red others Goblin-town spot stuff riddles hero?',
  },
  argTypes: {
    children: { control: 'array' },
  },
}

export const Default = Template.bind({})
