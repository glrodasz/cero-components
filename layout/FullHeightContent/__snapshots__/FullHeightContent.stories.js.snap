// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/FullHeightContent Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="full-height-content is-visible"
  >
    <div
      className="content"
    >
      Mice defiling required Bilbo sailing unspoiled this hunting. 14th refuge knowing breath Middle-earth sack demon children?
    </div>
    <div
      className="spacer horizontal"
      style={
        Object {
          "height": 15,
        }
      }
    />
    <div
      className="footer"
    >
      Ambition living aid indeed mutton suffered l failing kindness Ã‰othain? Red others Goblin-town spot stuff riddles hero?
    </div>
  </div>
</div>
`;
